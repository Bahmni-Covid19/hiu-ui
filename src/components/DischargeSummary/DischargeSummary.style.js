import {makeStyles} from "@material-ui/core/styles";

const useStyles = makeStyles((theme) => ({
    container: {
        backgroundColor: '#c8c8c8',
        padding: theme.spacing(1,2),
        textTransform:'uppercase',
        borderTopLeftRadius:theme.spacing(0.625),
        fontSize:'0.875rem',
        marginBottom: theme.spacing(0.25)
    },
    description: {
        textTransform:'none',
        marginLeft:theme.spacing(4),
        fontWeight: '400',
        fontSize: '0.875rem'
    }
}));

export default useStyles;