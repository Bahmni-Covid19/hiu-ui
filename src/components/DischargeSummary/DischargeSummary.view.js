import React from 'react';
import * as PropTypes from 'prop-types';
import Typography from "@material-ui/core/Typography";
import useStyle from './DischargeSummary.style';
import {formatDateString} from "../common/HealthInfo/FhirResourcesUtils";

const DischargeSummary = ({ title, startDate, endDate }) => {
    const classes  = useStyle();
    const start = formatDateString(startDate);
    const end = formatDateString(endDate);
    return(
        title ? (
          <Typography variant='h6' color='inherit' className={classes.container}>
            {title}
            { start && end && (
            <span className={classes.description}>
              Admitted from
              {' '}
              {start}
              {' '}
              to
              {' '}
              {end}
            </span>
              )}
          </Typography>
    ) : ''
    );
};

DischargeSummary.propTypes = {
    title: PropTypes.string,
    startDate: PropTypes.string,
    endDate: PropTypes.string,
};

DischargeSummary.defaultProps = {
    title: '',
    startDate: '',
    endDate: '',
};


export default DischargeSummary;