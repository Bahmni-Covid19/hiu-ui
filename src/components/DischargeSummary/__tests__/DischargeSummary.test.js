import React from 'react';
import {shallow} from "enzyme";
import {Typography} from "@material-ui/core";
import DischargeSummary from "../DischargeSummary.view";

describe('DischargeSummary', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(
          <DischargeSummary
            title="Discharge Summary"
            startDate="2018-04-01T15:30:10+01:00"
            endDate="2018-04-10T15:30:10+01:00"
          />
        );
    });

    it('should render properly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should have default props', () => {
        expect(DischargeSummary.defaultProps.title).toEqual('');
        expect(DischargeSummary.defaultProps.startDate).toEqual('');
        expect(DischargeSummary.defaultProps.endDate).toEqual('');
    });

    it('should not render description if startDate is empty', () => {
        wrapper.setProps({startDate: ''});
        expect(wrapper.find('span')).toHaveLength(0);
    });

    it('should not render description if endDate is empty', () => {
        wrapper.setProps({endDate: ''});
        expect(wrapper.find('span')).toHaveLength(0);
    });

    it('should not render anything if title is empty', () => {
        wrapper.setProps({title:''});
        expect(wrapper.find(Typography)).toHaveLength(0);
    });
});