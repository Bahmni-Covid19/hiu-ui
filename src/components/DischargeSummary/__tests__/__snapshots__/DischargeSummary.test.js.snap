// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DischargeSummary should render properly 1`] = `
<WithStyles(ForwardRef(Typography))
  className="makeStyles-container-1"
  color="inherit"
  variant="h6"
>
  Discharge Summary
  <span
    className="makeStyles-description-2"
  >
    Admitted from
     
    1/4/2018
     
    to
     
    10/4/2018
  </span>
</WithStyles(ForwardRef(Typography))>
`;
